src/sdam/description/topology/mod.rs

```diff
@@ -661,8 +666,13 @@ impl TopologyDescription {
     }
 
     /// Create a new ServerDescription for each address and add it to the topology.
-    fn add_new_servers<'a>(&mut self, servers: impl Iterator<Item = &'a String>) -> Result<()> {
-        let servers: Result<Vec<_>> = servers.map(|server| StreamAddress::parse(server)).collect();
+    fn add_new_servers<'a>(
```

Comment:
Rustfmt changed the spacing a bit here, so it might not be super clear what the change is: the return value of the function was changed from `Result<()>` to `Result<(), String>`, and then a call to `to_string()` on the error was added to the iterator expression.